src/nxt_h1proto.c

```diff
@@ -1824,6 +1836,12 @@ nxt_h1p_conn_sent(nxt_task_t *task, void *obj, void *data)
 {
     nxt_conn_t          *c;
     nxt_event_engine_t  *engine;
+#if (NXT_HAVE_OTEL)
+    nxt_http_request_t  *r;
+
+    r = ((nxt_h1proto_t *) data)->request;
+    nxt_otel_test_and_call_state(task, r);
```

Comment:
I wonder what happens if `nxt_h1p_conn_sent()` is called multiple times per request?

Reply:
the first call will trigger the span to be processed. All subsequent calls will do nothing to otel because the opentelemetry state machine will have shifted back to unset state.

It is completely safe.

Reply:
Is your intention to do the final step collection?
If yes, is it better to call it when the request is closed?